### PR TITLE
Improve Docstrings in Task/braket_simulator

### PR DESCRIPTION
**Pull request by shubhusion**
Fixes #957

### CODE REVIEW

File: src/bloqade/task/braket_simulator.py [hunk 1]
```diff
@@ -19,17 +24,48 @@
...
+        """
+        Runs the task on the Braket LocalSimulator.
+
+        Args:
+            **kwargs: Additional arguments for the simulator run method.
+
+        Returns:
+            BraketEmulatorTask: The current instance with the task results updated.
+
+        Example:
```
Comment (Roger-luo) [inline src/bloqade/task/braket_simulator.py:62]: same as my previous comment, this needs to use a markdown code block

Comment (Roger-luo) [reply src/bloqade/task/braket_simulator.py:62]: also can you actually run these code and copy the important printings back, e.g the printing of task and result

File: src/bloqade/task/braket_simulator.py [hunk 2]
```diff
@@ -38,14 +74,42 @@ def run(self, **kwargs) -> "BraketEmulatorTask":
...
+        Retrieves the results of the task.
+
+        Returns:
+            QuEraTaskResults: The results of the task.
+
+        Raises:
+            ValueError: If the task has not been submitted yet.
+
+        Example:
+            >>> task_spec = BraketTaskSpecification(...)  # Task specification setup
```
Comment (Roger-luo) [inline src/bloqade/task/braket_simulator.py:88]: this needs to be more specific, e.g what does the specification take? and I don't think this is what user will be using. The result is called in the builder call chain.

File: src/bloqade/task/braket_simulator.py [hunk 3]
```diff
@@ -38,14 +74,42 @@ def run(self, **kwargs) -> "BraketEmulatorTask":
...
-    def nshots(self):
+    def nshots(self) -> int:
+        """
+        Returns the number of shots for the task.
+
+        Returns:
+            int: The number of shots.
+
+        Example:
+            >>> task_spec = BraketTaskSpecification(...)  # Task specification setup
```
Comment (Roger-luo) [inline src/bloqade/task/braket_simulator.py:108]: same, you can't ask a user to construct task specification directly

**Review by Roger-luo — CHANGES_REQUESTED** [review]
I think in general I think you need to run the emulator to come up with better examples, e.g demonstrate what intermediate result people will get by calling `run`? what next steps they can do after calling `run`? I also leave a few more specific comment above. I don't think you should have things like metadata show up in an example that is not very idiomatic.

File: src/bloqade/task/braket_simulator.py [hunk 4]
```diff
@@ -19,17 +24,48 @@
...
+        metadata (Dict[str, ParamType]): Metadata related to the task.
+        task_result_ir (Optional[QuEraTaskResults]): The results of the task, if available.
+    """
+
     task_ir: BraketTaskSpecification
     metadata: Dict[str, ParamType]
     task_result_ir: Optional[QuEraTaskResults] = None
 
     def _geometry(self) -> Geometry:
+        """
```
Comment (Roger-luo) [inline src/bloqade/task/braket_simulator.py:41]: This is a private function, and I'm doing duplicated reviews on the same issue. Please work on one of your PRs first before requesting multiple reviews thank you.

File: src/bloqade/task/braket_simulator.py (hunk 1 above)
Comment (weinbe58) [reply src/bloqade/task/braket_simulator.py:62]: Just for clarity you just need to do:

```python
"""

Example:
    ```python
    >>> <example code>
    >>> ...
    ```

"""
```

**Pull request closed without merging**